File: app/src/main/java/com/example/cs_topics_project_test/function_java/Time.java
package com.example.cs_topics_project_test.function_java;

import androidx.annotation.NonNull;

public class Time {
    private int hour = 0;
    private final int min;
    private final boolean isAm;
    public Time(int hour, int min, boolean isAm) {
        this.min = min % 60;
        if (min > 60) {
            hour += min / 60;
        }
        if (hour > 24) {
            throw new IllegalArgumentException("Invalid Time entered");
        }
        this.hour = hour % 12;
        if (hour >= 12) {
            this.isAm = false;
        } else {
            this.isAm = true;
        }
    }

    public int getHour() {
        return this.hour;
    }

    public int getMin() {
        return this.min;
    }

    public boolean isAm() {
        return this.isAm;
    }

    public boolean isPm() {
        return !this.isAm;
    }

    @NonNull
    @Override
    public String toString() {
        return this.hour + ":" + this.min;
    }
}

File: app/src/main/java/com/example/cs_topics_project_test/function/Time.kt
package com.example.cs_topics_project_test.function

class Time(hour: Int, min: Int, second: Int) {
    private val hour: Int
    private val min: Int
    private val second: Int

    init {
        if (hour > 24) throw IllegalArgumentException("invalid hour as input")
        else this.hour = hour
        if (min > 60) throw IllegalArgumentException("invalid minute as input")
        else this.min = min
        if (second > 60) throw IllegalArgumentException("invalid minute as input")
        else this.second = second
    }

    fun getHour(): Int {
        return this.hour
    }

    fun getMin(): Int {
        return this.min
    }

    fun getSecond(): Int {
        return this.second
    }

    override fun toString(): String {
        return this.hour.toString() + ":" + this.min + ":" + this.second
    }
}